### PR TITLE
fetch bills

### PR DESCRIPTION
**Pull request by OLI9292**

### CODE REVIEW

File: Track Your Reps/Views/FeedTableCell.swift [hunk 1]
```diff
@@ -13,7 +13,7 @@ class FeedTableCell: UITableViewCell {
     
     var event: Event? {
         didSet {
-            eventQuestionLabel.text = event?.question
+            eventQuestionLabel.text = "\(event!.isBill ? "I'm a bill! - " : "")\(event?.question)"
```
Comment (OLI9292) [inline Track Your Reps/Views/FeedTableCell.swift:16]: just an example of using bill data in the feed table

`.isBill` is an easy computed var helper to determine whether an event is/has a bill

File: Track Your Reps/TabBarController.swift [hunk 2]
```diff
@@ -25,11 +28,10 @@ class TabBarController: UITabBarController, UITabBarControllerDelegate {
     
     func setupUser() {
         guard let userState = userState else { onboardUser(); return }
+        User.sharedInstance.delegate = self
         User.sharedInstance.state = userState
         User.sharedInstance.fetchMembers()
-        User.sharedInstance.fetchEvents() {
-            self.setupViewControllers()
```
Comment (OLI9292) [inline Track Your Reps/TabBarController.swift:31]: so no closure anymore to setup VCs, we'll do this via the SetupViewControllersDelegate

File: Track Your Reps/Models/User.swift [hunk 3]
```diff
@@ -73,12 +75,49 @@ extension User {
...
+        
+        group.notify(queue: DispatchQueue.main, execute: {
+            self.dataStore.members = self.dataStore.members.map { member in
+                return self.resetEvents(for: member, eventsWithBills: withBills)
+            }
+            self.delegate?.setupViewControllers()
+        })
+    }
+    
+    fileprivate func resetEvents(for member: CongressMember, eventsWithBills withBills: [Event]) -> CongressMember {
```
Comment (OLI9292) [inline Track Your Reps/Models/User.swift:114]: reminder to myself: think I can shorten these resetEvents methods with a tip from ProSwift

File: Track Your Reps/Lib/ResponseStubs.swift [hunk 4]
```diff
@@ -12,7 +12,9 @@ public func stubbedResponse(_ stub: ResponseStub) -> Data {
 
 public enum ResponseStub: String {
     
-    case votesForMember = "{\"status\": \"OK\", \"copyright\": \"Pro Publica\", results: [{\"member_id\": \"1\", \"total_votes\": \"100\", \"offset\": \"0\", \"votes\": [{\"member_id\": \"1\", \"chamber\": \"Senate\", \"congress\": \"115\", \"session\": \"1\", \"roll_call\": \"1\", \"bill\": {\"number\": \"Res. 37\", \"bill_uri\": \"api.propublica.org/congress/v1/115/bills/hjres37\", \"title\": \"Hello title hello\", \"latest_action\": \"So much action\"}, \"member_id\": \"1\", \"description\": \"Something something something.\", \"question\": \"Is a question a question?\", \"date\": \"2017-03-02\", \"time\": \"0:00\", \"position\": \"Yes\"}]}]}"
+    case bill = "{}"
```
Comment (OLI9292) [inline Track Your Reps/Lib/ResponseStubs.swift:15]: **Todo**

add responseStub for bill and add a helper to the API wrapper so these fire when working offline allowing us to work w/o internet

**Comment by bencbernstein** [comment]
excellent

File: Track Your Reps/TabBarController.swift (hunk 2 above)
Comment (bencbernstein) [reply Track Your Reps/TabBarController.swift:31]: it all comes full circle

**Pull request merged**